src/infrastructure/jira/jira-service.ts

```diff
@@ -247,31 +245,38 @@ class JiraService {
 	 */
 	updateAttachedDesignUrlV2IssueProperty = async (
 		issueIdOrKey: string,
-		{ url, displayName }: AtlassianDesign,
+		design: AtlassianDesign,
 		connectInstallation: ConnectInstallation,
 	): Promise<void> => {
-		const storedValue =
-			await this.getIssuePropertyJsonValue<AttachedDesignUrlV2IssuePropertyValue>(
-				issueIdOrKey,
-				issuePropertyKeys.ATTACHED_DESIGN_URL_V2,
-				connectInstallation,
-				ATTACHED_DESIGN_URL_V2_VALUE_SCHEMA,
-			);
+		const storedValue = await this.getIssuePropertyJsonValue<
+			AttachedDesignUrlV2IssuePropertyValue[]
+		>(
+			issueIdOrKey,
+			issuePropertyKeys.ATTACHED_DESIGN_URL_V2,
+			connectInstallation,
+			ATTACHED_DESIGN_URL_V2_VALUE_SCHEMA,
+		);
 
-		const isTargetValueItemStored = storedValue?.some(
-			(item) => item.url === url && item.name === displayName,
+		const storedItem = storedValue?.find((item) =>
+			this.areUrlsOfSameDesign(item.url, design.url),
 		);
 
-		if (isTargetValueItemStored) return;
+		const newItem = {
+			url: JiraService.buildDesignUrlForIssueProperties(design),
+			name: design.displayName,
+		};
 
-		const storedValueExcludingItemWithTargetUrl =
-			storedValue?.filter((item) => !this.areUrlsOfSameDesign(item.url, url)) ??
-			[];
+		if (storedItem?.url === newItem.url && storedItem?.name === newItem.name) {
+			return;
+		}
 
-		const newValue = [
-			...storedValueExcludingItemWithTargetUrl,
-			{ url, name: displayName },
-		];
+		let newValue = storedValue ? [...storedValue] : [];
+
+		if (storedItem) {
+			newValue = newValue.map((item) => (item === storedItem ? newItem : item));
```

Comment:
When the design already exists in the issue property, you decided to update it with the new value? Just confirming this is correct because we were discussing leaving the original URL unchanged and perhaps only updating the name.

Reply:
Yes, sorry, I had to explain this on the PR.

This was my initial plan and implementation, but, unfortunately, it didn't fix the issue and I had to change the approach.

Please, consider the following scenario with the initially planned solution:

1. A user adds a design from the old experience
   ```js
   // Issue Properties
   {
      "attached-design-url-v2": [
        {
          url: "https://www.figma.com/file/abc/Cool-Design?mode=design",
          name: "Cool Design"
        }
      ]
   }
   ```
 2. A user switches to the new experience and backfills data:
   ```js
   // Issue Properties
   {
      "attached-design-url": "https://www.figma.com/file/abc", // Normalised
      "attached-design-url-v2": [
        {
          url: "https://www.figma.com/file/abc/Cool-Design?mode=design", // Original
          name: "Cool Design"
        }
      ]
   }
   ```  
 3. A user switches back to the old experience:
    ```js
   // Issue Properties
   {
      "attached-design-url": "https://www.figma.com/file/abc",
      "attached-design-url-v2": [
        {
          url: "https://www.figma.com/file/abc/Cool-Design?mode=design", // Original
          name: "Cool Design"
        },
        {
          url: "https://www.figma.com/file/abc", // Ooops! Added from `attached-design-url`.
          name: "Cool Design"
        }
      ]
   }
   ```

Therefore, while we write to `attached-design-url`, we have to write it **consistently** with `attached-design-url-v2`, so the same URL is written to both Issue Properties. This is what the PR addresses.

As we discussed, there is still an edge case when a single design can be duplicated (`attached-design-url`):

1. A user links a design using the "Jira" widget in Figma 
2. A switches from the new experience to the old one

There is a cure for this issue but it would make the logic complicated, so it might be better to avoid it. It is an edge case and, normally, should never affect our users (ideally, we should not switch back to the old experience).